internal/config/config.go

```diff
@@ -109,22 +118,28 @@ func strOrDefault(key, d string) string {
 
 func durationOrDefault(key string, d time.Duration) time.Duration {
 	str := strOrDefault(key, "nan")
+	if str == "nan" {
```

Comment:
why here we are considering empty env var value as a "valid" value which we try to parse and in `boolOrDefault` we consider an empty value in the same way as if the env var was not defined

Examples:
- `REPORTING_RATE=""` will fail to parse
- `FAST_INGEST=""` will use the default and set it to `true`